### DISCUSSION TITLE
Question about rom banks

### DISCUSSION THREAD

**Discussion by LowEnergeticAg** [vhelin/wla-dx#486] [Q&A]
I recently found this compiler when I looked for 65816 assembler as I build my own computer based on this cpu on breadboards. I plan to expand the memory map as I get more peripherals that I can connect to my computer. The current memory map looks like this:
0x000000-0x007FFF - SRAM
0x008000-0x00BFFF - LCD display (temporary to make sure computer is working)
0x00C000-0x00FFFF - EEPROM (functions as a rom)
0x010000-0xFFFFFF - unused
I already know that the computer is working as i wrote a simple test program and compiled it using VASM, but it only supports 6502 so i was limited to 8 bit mode. To try out wla I wrote a simple test program:

```
.MEMORYMAP
DEFAULTSLOT 1
SLOTSIZE $8000
SLOT 0 $0000
SLOTSIZE $4000
SLOT 1 $COOO
.ENDME

.ROMBANKMAP
BANKSTOTAL 2
BANKSIZE $C000
BANKS 1
BANKSIZE $4000
BANKS 1
.ENDRO

.BANK 1
.ORG $C000
.ACCU 8
.INDEX 8
lda #$00

```
However, when I try to compile it with this command:
`wla-65816 test.asm`
I get an error:
`MEM_INSERT: Origin ($c000) overflows from bank (1).
   ^ mathlib.asm:0: Writing a byte`

I tried to read the manual, but after reading about rom maps and memory maps for a few times, I think I won't be able to solve this on my own. If you know what am I doing wrong, please reply.

**Comment by vhelin** [comment]
.ORG is offset from the slot's starting address, .ORGA is the address in memory.

I don't quite understand your .ROMBANKMAP, it has two banks, 1st is $C000 bytes in size and 2nd is $4000 bytes in size, but your .MEMORYMAP suggests slot 0 is $8000 bytes in size and slot 1 is $4000 bytes in size. Your ROM bank 1 doesn't fit into those slots.

I'm guessing that you want to write data into the beginning of ROM bank 1 and have that in slot 1? In that case change your .ORG there to .ORGA or make it ".ORG $0000"... ? EDIT: Also say ".BANK 1 SLOT 1" instead of just ".BANK 1"...

**Comment by LowEnergeticAg** [comment]
Thanks for the reply
You're right, I wanted to write to the start off the rom and misunderstod the .ORG directive.
I also specified size od the first bank to $c000 because i though that rombanks need to occupy continous spacer on menory after setting it to $8000 resultet on the same error because i misunderstod .ORG directive.
